app/controllers/cards_controller.rb

```diff
@@ -2,4 +2,48 @@ class CardsController < ApplicationController
   def index
     @cards = Card.all
   end
+
+  def show
+    @card = Card.find(params[:id])
+  end
+
+  def new
+    @card = Card.new
+  end
+
+  def edit
+    @card = Card.find(params[:id])
+  end
+ 
+  def create
+    @card = Card.new(card_params)
+ 
+    if @card.save
+      redirect_to @card
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    @card = Card.find(params[:id])
```

Comment:
you are using this line so much times, time to before_action! 
http://guides.rubyonrails.org/action_controller_overview.html#filters

P.S. this is related to question about DRY in your task.